libs/template/experimental/virtual-scrolling/src/lib/scroll-strategies/fixed-size-virtual-scroll-strategy.ts

```diff
@@ -0,0 +1,278 @@
+                index,
+              });
+            })
+          );
+        }),
+        this.until$
+      )
+      .subscribe();
+  }
+
+  private calcRenderedRange(): void {
+    const dataLengthChanged$ = this.viewRepeater.values$.pipe(
+      map(
+        (values) =>
+          (Array.isArray(values)
+            ? values
+            : values != null
+            ? Array.from(values)
+            : []
+          ).length
+      ),
+      distinctUntilChanged(),
+      shareReplay({ bufferSize: 1, refCount: true })
+    );
+    dataLengthChanged$.subscribe((dataLength) => {
+      this.contentSize = dataLength * this.itemSize;
+    });
+    const onScroll$ = this.viewport.elementScrolled$.pipe(
+      map(() => this.viewport.getScrollTop()),
+      startWith(0),
+      tap((_scrollTop) => {
+        this.direction = _scrollTop > this.scrollTop ? 'down' : 'up';
+        this.scrollTop = _scrollTop;
+      })
+    );
+    combineLatest([dataLengthChanged$, this.viewport.containerSize$, onScroll$])
+      .pipe(
+        map(([length, containerSize]) => {
+          const range = { start: 0, end: 0 };
+          if (this.direction === 'up') {
+            range.start = Math.floor(
+              Math.max(0, this.scrollTop - this.runwayItems * this.itemSize) /
+                this.itemSize
+            );
+            range.end = Math.min(
+              length,
+              Math.ceil(
+                (this.scrollTop +
+                  containerSize +
+                  this.runwayItemsOpposite * this.itemSize) /
+                  this.itemSize
+              )
+            );
+          } else {
+            range.start = Math.floor(
+              Math.max(
+                0,
+                this.scrollTop - this.runwayItemsOpposite * this.itemSize
+              ) / this.itemSize
+            );
+            range.end = Math.min(
+              length,
+              Math.ceil(
+                (this.scrollTop +
+                  containerSize +
+                  this.runwayItems * this.itemSize) /
+                  this.itemSize
+              )
+            );
+          }
+          this.scrolledIndex = Math.floor(this.scrollTop / this.itemSize);
+          return range;
+        }),
+        distinctUntilSomeChanged(['start', 'end']),
+        this.until$
+      )
+      .subscribe((range: ListRange) => (this.renderedRange = range));
+  }
+
+  scrollToIndex(index: number, behavior?: ScrollBehavior): void {
+    const scrollTop = this.itemSize * index;
+    this.viewport.scrollTo(scrollTop, behavior);
+  }
+
+  private _setViewPosition(
+    view: EmbeddedViewRef<RxVirtualForViewContext<T, U>>,
+    scrollTop: number
+  ): void {
+    const element = this.getElement(view);
+    element.style.position = 'absolute';
+    element.style.transform = `translateY(${scrollTop}px)`;
+    if (this.enterStyles) {
+      Object.keys(this.enterStyles).forEach((style) => {
+        element.style[style] = this.enterStyles[style];
```

Comment:
enterStyles are not working as suggested by the JsDoc at the top. Both the CSS styling (opacity: 0) as well as the enterStyles (opacity: 1) are set in the same frame. Therefore, the CSS transition does not trigger any animation (tested in Chrome 107). Could be solved be executing this code-block in a raf. :) 

Nevertheless, nice work!